api-server.py

```diff
@@ -140,8 +185,57 @@ def main():
         endpoint_rest = "tcp:port={0}:interface={1}".format(args.port_rest, host)
         endpoints.serverFromString(reactor, endpoint_rest).listen(Site(api_server_rest.app.resource()))
 
-    app = Klein()
-    app.run(host, 9999)
+    app = ApiKlein()
+    app.run(host, 9999, syslog_facility=syslog_facility)
+
+    # After the reactor is stopped, gracefully shutdown the database.
+    logger.info("Shutting down - closing databases...")
+    NotificationDB.close()
+    Blockchain.Default().Dispose()
+    NodeLeader.Instance().Shutdown()
+
+
+class ApiKlein(Klein):
+    """
+    ApiKlein extends Klein so that the logging behavior can be customized. Aside from logging,
+    the implementation is identical to Klein.run(): https://github.com/twisted/klein/blob/master/src/klein/_app.py#L376
+    """
+    def run(self, host=None, port=None, endpoint_description=None, syslog_facility=None):
+        if syslog_facility is not None:
+            facility = translate_syslog_facility(syslog_facility)
+            startLogging(prefix="pyapi", facility=facility)
```

Comment:
With the approach you have here now, we will only call `startLogging` when logging to `syslog`. Shouldn't we be calling `startLoggingWithObserver` and pass in the `observer` created above? I'm actually wondering if we do that, do we even need all of this specialized `syslog_facility` and `translate_syslog_facility` at all? If the `PythonLoggingObserver` handles unifying the logging, it seems like that could unify the logs entirely and simplify this quite a bit?

Reply:
I think it would be a good approach to setup logzero to use the SyslogHandler, and just have the same setup for Twisted everywhere, to simply use the logzero logger setup.